src/mds/Server.cc

```diff
@@ -130,6 +130,9 @@ void Server::dispatch(Message *m)
 	  dout(3) << "queuing completed op" << dendl;
 	  queue_replay = true;
 	}
+	// this request was created before the cap reconnect message, drop any embedded
+	// cap releases.
+	req->releases.clear();
```

Comment:
The hammer client isn't behaving properly right now, which might mean we want to have the MDS clear them out. I'm not sure if we want to do that server-side though since it means we can't make clients cleverer in the future. But the kernel client might have the same issue?

In any case we should also backport this to hammer, so please tag the commit for that @ukernel.